### PR TITLE
fix for kinematic constraints parsing

### PR DESCRIPTION
**Pull request by jrgnicho**
### Description

Minor change to properly parse the constraints parameter loaded from a yaml file.  As described in the `kinematic_constraints::constructConstraints` function the yaml file needs to have the following structure:
```
 name: constraint_name
 constraint_ids: [constraint_1, constraint_2]
 constraints:
   constraint_1:
     type: orientation
     frame_id: world
     link_name: tool0
     orientation: [0, 0, 0]  # [r, p, y]
     tolerances: [0.01, 0.01, 3.15]
     weight: 1.0
   constraint_2:
     type: position
     frame_id: base_link
     link_name: tool0
     target_offset: [0.1, 0.1, 0.1]  # [x, y, z]
     region:
       x: [0.1, 0.4]  # [min, max]
       y: [0.2, 0.3]
       z: [0.1, 0.6]
     weight: 1.0
```
Prior to this change this yaml structure fails to load.
### Checklist
- [ x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ N/A] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ N/A] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ N/A] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ N/A]  Include a screenshot if changing a GUI
- [ x] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by sea-bass — CHANGES_REQUESTED** [review]
I've traced down all the dots and parameter structure and this LGTM.

However, two things:
1. Could you rebase and retarget this to the `main` branch so the fix goes there, and then I can mark this PR for humble backporting?
2. I'm actually not a maintainer, so we'll still need other approvals :)

**Review by sjahr — APPROVED** [review]
Looks good, thanks for the fix! If you re-target it at main, we can easily backport it by just adding and additional tag

**Comment by sea-bass** [comment]
> OK, I can do that. For future PR's targeting a branch other than main how should I proceed?

I think what you just did is perfect: Put up PR to main, add the `backport-humble` label, and that should do it :)

The one exception would be if the PR in question is a bug fix that doesn't make sense on `main` (it's been fixed/refactored, etc.) in which case targeting the non-main branch directly is OK.

**Review by sea-bass — APPROVED** [review]

**Pull request merged**
